query-engine/core/src/interpreter/query_interpreters/read.rs

```diff
@@ -107,6 +122,12 @@ fn read_many(
             .await?;
 
         let scalars = if let Some(p) = processor {
+            let scalars = if req_inmem_distinct {
+                p.apply_distinct(scalars)
+            } else {
+                scalars
+            };
```

Comment:
I might be missing something but it feels like this bit of logic could be put in `.apply` without leaking it outside. Are there any places where `.apply` is called and we don't want to do this logic?